src/kernels/jupyter/connection/jupyterServerProviderRegistry.ts

```diff
@@ -262,9 +274,9 @@ class JupyterUriProviderAdaptor extends Disposables implements IJupyterUriProvid
         if (!this.provider.serverProvider) {
             throw new Error(`No Jupyter Server Provider for ${this.provider.extensionId}#${this.provider.id}`);
         }
-        const servers = await this.provider.serverProvider.provideJupyterServers(token);
+        const servers = await Promise.resolve(this.provider.serverProvider.provideJupyterServers(token));
```

Comment:
its possible its a sync return value, it’s a method that’s implemented by 3rd party extensions
 simple trick to avoid checking if return value is a promise or not
We do this in core as well for apis